src/database/models/Product.ts

```diff
@@ -0,0 +1,105 @@
+import { Model, DataTypes } from 'sequelize';
+import sequelize from './index';
+import { UUIDV4 } from 'sequelize';
+import { Size } from './Size';
+import { Category } from './Category';
+
+export interface ProductAttributes {
+  id?: number;
+  sellerId: string;
+  name: string;
+  description: string;
+  images: string[];
+  color?: string[];
+  discount?: number;
+  price?: number;
+  quantity?: number;
+  expiryDate?: Date;
+  categoryId: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+}
+
+export class Product extends Model<ProductAttributes> implements ProductAttributes {
+  public id!: number;
+  public sellerId!: string;
+  public name!: string;
+  public description!: string;
+  public discount!: number;
+  public quantity!: number;
+  public price!: number;
+  public expiryDate!: Date;
+  public categoryId!: string;
+  public images!: string[];
+  public color!: string[];
+  public readonly createdAt!: Date | undefined;
+  public readonly updatedAt!: Date | undefined;
+}
+
+Product.init(
+  {
+    id: {
+      type: DataTypes.UUID,
+      defaultValue: UUIDV4,
+      primaryKey: true,
+      unique: true,
+    },
+    name: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    description: {
+      type: DataTypes.TEXT,
+      allowNull: false,
+    },
+    color: {
+      type: DataTypes.ARRAY(DataTypes.STRING),
+      allowNull: true,
+    },
+    discount: {
+      type: DataTypes.FLOAT,
+      allowNull: true,
+      defaultValue: 0,
+    },
+    price: {
```

Comment:
what is the difference between this price here and the one in size model? i think this can be removed. we can use price which is in size model cz every size can have it own price.